// @kontent-ai/management-sdk/dist/es6/responses/taxonomies/taxonomy-responses.js
import { BaseResponses } from '../base-responses';
export var TaxonomyResponses;
(function (TaxonomyResponses) {
    class TaxonomyListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.TaxonomyListResponse = TaxonomyListResponse;
    class ListAllTaxonomiesResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    TaxonomyResponses.ListAllTaxonomiesResponse = ListAllTaxonomiesResponse;
    class GetTaxonomyResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.GetTaxonomyResponse = GetTaxonomyResponse;
    class AddTaxonomyResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.AddTaxonomyResponse = AddTaxonomyResponse;
    class ModifyTaxonomyResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.ModifyTaxonomyResponse = ModifyTaxonomyResponse;
})(TaxonomyResponses || (TaxonomyResponses = {}));
