// @kontent-ai/management-sdk/dist/cjs/responses/content-items/content-item-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentItemResponses = void 0;
const base_responses_1 = require("../base-responses");
var ContentItemResponses;
(function (ContentItemResponses) {
    class ContentItemsResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.ContentItemsResponse = ContentItemsResponse;
    class ContentItemsListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    ContentItemResponses.ContentItemsListAllResponse = ContentItemsListAllResponse;
    class ViewContentItemResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.ViewContentItemResponse = ViewContentItemResponse;
    class AddContentItemResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.AddContentItemResponse = AddContentItemResponse;
    class UpdateContentItemResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.UpdateContentItemResponse = UpdateContentItemResponse;
    class UpsertContentItemResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.UpsertContentItemResponse = UpsertContentItemResponse;
})(ContentItemResponses || (exports.ContentItemResponses = ContentItemResponses = {}));
