// @kontent-ai/management-sdk/dist/cjs/responses/taxonomies/taxonomy-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxonomyResponses = void 0;
const base_responses_1 = require("../base-responses");
var TaxonomyResponses;
(function (TaxonomyResponses) {
    class TaxonomyListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.TaxonomyListResponse = TaxonomyListResponse;
    class ListAllTaxonomiesResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    TaxonomyResponses.ListAllTaxonomiesResponse = ListAllTaxonomiesResponse;
    class GetTaxonomyResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.GetTaxonomyResponse = GetTaxonomyResponse;
    class AddTaxonomyResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.AddTaxonomyResponse = AddTaxonomyResponse;
    class ModifyTaxonomyResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    TaxonomyResponses.ModifyTaxonomyResponse = ModifyTaxonomyResponse;
})(TaxonomyResponses || (exports.TaxonomyResponses = TaxonomyResponses = {}));
