// @kontent-ai/management-sdk/dist/esnext/models/content-management-api-endpoints.js
export class ContentManagementApiEndpoints {
    constructor(data) {
        this.projectsPath = 'projects';
        this.subscriptionsPath = 'subscriptions';
        this.environmentId = data.environmentId;
        this.subscriptionId = data.subscriptionId;
    }
    changeWorkflowStepOfLanguageVariant(itemIdentifier, languageIdentifier, workflowIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/workflow/${workflowIdentifier.getParamValue()}`;
    }
    publishLaguageVariant(itemIdentifier, languageIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/publish`;
    }
    createNewVersionOfALanguageVariant(itemIdentifier, languageIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/new-version`;
    }
    unpublishLanguageVariant(itemIdentifier, languageIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/unpublish-and-archive`;
    }
    cancelScheduledPublishingOfLanguageVariant(itemIdentifier, languageIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/cancel-scheduled-publish`;
    }
    cancelScheduledUnpublishingOfLanguageVariant(itemIdentifier, languageIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/cancel-scheduled-unpublish`;
    }
    listWorkflowSteps() {
        return `${this.getEnvironmentsPath()}/workflow`;
    }
    listWorkflows() {
        return `${this.getEnvironmentsPath()}/workflows`;
    }
    addWorkflow() {
        return `${this.getEnvironmentsPath()}/workflows`;
    }
    updateWorkflow(workflowIdentifier) {
        return `${this.getEnvironmentsPath()}/workflows/${workflowIdentifier.getParamValue()}`;
    }
    deleteWorkflow(identifier) {
        return `${this.getEnvironmentsPath()}/workflows/${identifier.getParamValue()}`;
    }
    changeWorkflowOfLanguageVariant(itemIdentifier, languageIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${languageIdentifier.getParamValue()}/change-workflow`;
    }
    listContentTypeSnippets() {
        return `${this.getEnvironmentsPath()}/snippets`;
    }
    viewContentTypeSnippet(identifier) {
        return `${this.getEnvironmentsPath()}/snippets/${identifier.getParamValue()}`;
    }
    addContentTypeSnippet() {
        return `${this.getEnvironmentsPath()}/snippets`;
    }
    deleteContentTypeSnippet(identifier) {
        return `${this.getEnvironmentsPath()}/snippets/${identifier.getParamValue()}`;
    }
    modifyContentTypeSnippet(identifier) {
        return `${this.getEnvironmentsPath()}/snippets/${identifier.getParamValue()}`;
    }
    startProjectValidation() {
        return `${this.getEnvironmentsPath()}/validate-async`;
    }
    checkEnvironmentValidation(identifier) {
        return `${this.getEnvironmentsPath()}/validate-async/tasks/${identifier.getParamValue()}`;
    }
    listEnvironmentIssues(identifier) {
        return `${this.getEnvironmentsPath()}/validate-async/tasks/${identifier.getParamValue()}/issues`;
    }
    environmentInformation() {
        return `${this.getEnvironmentsPath()}`;
    }
    listContentTypes() {
        return `${this.getEnvironmentsPath()}/types`;
    }
    addContentType() {
        return `${this.getEnvironmentsPath()}/types`;
    }
    deleteContentType(identifier) {
        return `${this.getEnvironmentsPath()}/types/${identifier.getParamValue()}`;
    }
    viewContentType(identifier) {
        return `${this.getEnvironmentsPath()}/types/${identifier.getParamValue()}`;
    }
    modifyContentType(identifier) {
        return `${this.getEnvironmentsPath()}/types/${identifier.getParamValue()}`;
    }
    addCustomApp() {
        return `${this.getEnvironmentsPath()}/custom-apps`;
    }
    modifyCustomApp(identifier) {
        return `${this.getEnvironmentsPath()}/custom-apps/${identifier.getParamValue()}`;
    }
    deleteCustomApp(identifier) {
        return `${this.getEnvironmentsPath()}/custom-apps/${identifier.getParamValue()}`;
    }
    getCustomApp(identifier) {
        return `${this.getEnvironmentsPath()}/custom-apps/${identifier.getParamValue()}`;
    }
    listCustomApps() {
        return `${this.getEnvironmentsPath()}/custom-apps`;
    }
    addTaxonomy() {
        return `${this.getEnvironmentsPath()}/taxonomies`;
    }
    modifyTaxonomy(identifier) {
        return `${this.getEnvironmentsPath()}/taxonomies/${identifier.getParamValue()}`;
    }
    deleteTaxonomy(identifier) {
        return `${this.getEnvironmentsPath()}/taxonomies/${identifier.getParamValue()}`;
    }
    listTaxonomies() {
        return `${this.getEnvironmentsPath()}/taxonomies`;
    }
    getTaxonomy(identifier) {
        return `${this.getEnvironmentsPath()}/taxonomies/${identifier.getParamValue()}`;
    }
    addAsset() {
        return `${this.getEnvironmentsPath()}/assets`;
    }
    deleteAsset(identifier) {
        return `${this.getEnvironmentsPath()}/assets/${identifier.getParamValue()}`;
    }
    upsertAsset(assetIdentifier) {
        return `${this.getEnvironmentsPath()}/assets/${assetIdentifier.getParamValue()}`;
    }
    uploadBinaryFile(filename) {
        return `${this.getEnvironmentsPath()}/files/${filename}`;
    }
    listAssets() {
        return `${this.getEnvironmentsPath()}/assets`;
    }
    viewAsset(identifier) {
        return `${this.getEnvironmentsPath()}/assets/${identifier.getParamValue()}`;
    }
    items() {
        return `${this.getEnvironmentsPath()}/items`;
    }
    viewContentItem(identifier) {
        return `${this.getEnvironmentsPath()}/items/${identifier.getParamValue()}`;
    }
    addContentItem() {
        return `${this.getEnvironmentsPath()}/items`;
    }
    updateContentItem(identifier) {
        return `${this.getEnvironmentsPath()}/items/${identifier.getParamValue()}`;
    }
    upsertContentItem(identifier) {
        return `${this.getEnvironmentsPath()}/items/${identifier.getParamValue()}`;
    }
    deleteContentItem(identifier) {
        return `${this.getEnvironmentsPath()}/items/${identifier.getParamValue()}`;
    }
    listLanguageVariantsOfItem(identifier) {
        return `${this.getEnvironmentsPath()}/items/${identifier.getParamValue()}/variants`;
    }
    listLanguageVariantsOfContentType(identifier) {
        return `${this.getEnvironmentsPath()}/types/${identifier.getParamValue()}/variants`;
    }
    listLanguageVariantsByCollection(identifier) {
        return `${this.getEnvironmentsPath()}/collections/${identifier.getParamValue()}/variants`;
    }
    listLanguageVariantsBySpace(identifier) {
        return `${this.getEnvironmentsPath()}/spaces/${identifier.getParamValue()}/variants`;
    }
    listLanguageVariantsOfContentTypeWithComponents(identifier) {
        return `${this.getEnvironmentsPath()}/types/${identifier.getParamValue()}/components`;
    }
    filterLanguageVariants() {
        return `${this.getEnvironmentsPath()}/early-access/variants/filter`;
    }
    deleteLanguageVariant(itemIdentifier, langaugeIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${langaugeIdentifier.getParamValue()}`;
    }
    viewLanguageVariant(itemIdentifier, langaugeIdentifier, isPublished) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${langaugeIdentifier.getParamValue()}${isPublished ? '/published' : ''}`;
    }
    upsertLanguageVariant(itemIdentifier, langaugeIdentifier) {
        return `${this.getEnvironmentsPath()}/items/${itemIdentifier.getParamValue()}/variants/${langaugeIdentifier.getParamValue()}`;
    }
    viewLanguage(identifier) {
        return `${this.getEnvironmentsPath()}/languages/${identifier.getParamValue()}`;
    }
    addLanguage() {
        return `${this.getEnvironmentsPath()}/languages`;
    }
    modifyLanguage(identifier) {
        return `${this.getEnvironmentsPath()}/languages/${identifier.getParamValue()}`;
    }
    listLanguages() {
        return `${this.getEnvironmentsPath()}/languages`;
    }
    deleteLegacyWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks/${identifier.getParamValue()}`;
    }
    getLegacyWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks/${identifier.getParamValue()}`;
    }
    addLegacyWebhook() {
        return `${this.getEnvironmentsPath()}/webhooks`;
    }
    enableLegacyWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks/${identifier.getParamValue()}/enable`;
    }
    disableLegacyWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks/${identifier.getParamValue()}/disable`;
    }
    listLegacyWebhooks() {
        return `${this.getEnvironmentsPath()}/webhooks`;
    }
    deleteWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks-vnext/${identifier.getParamValue()}`;
    }
    getWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks-vnext/${identifier.getParamValue()}`;
    }
    addWebhook() {
        return `${this.getEnvironmentsPath()}/webhooks-vnext`;
    }
    enableWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks-vnext/${identifier.getParamValue()}/enable`;
    }
    disableWebhook(identifier) {
        return `${this.getEnvironmentsPath()}/webhooks-vnext/${identifier.getParamValue()}/disable`;
    }
    listWebhooks() {
        return `${this.getEnvironmentsPath()}/webhooks-vnext`;
    }
    listAssetFolders() {
        return `${this.getEnvironmentsPath()}/folders`;
    }
    addAssetFolders() {
        return `${this.getEnvironmentsPath()}/folders`;
    }
    modifyAssetFolders() {
        return `${this.getEnvironmentsPath()}/folders`;
    }
    listCollections() {
        return `${this.getEnvironmentsPath()}/collections`;
    }
    setCollections() {
        return `${this.getEnvironmentsPath()}/collections`;
    }
    listAssetRenditions(identifier) {
        return `${this.getEnvironmentsPath()}/assets/${identifier.getParamValue()}/renditions`;
    }
    addAssetRendition(identifier) {
        return `${this.getEnvironmentsPath()}/assets/${identifier.getParamValue()}/renditions`;
    }
    modifyAssetRendition(assetIdentifier, renditionIdentifier) {
        return `${this.getEnvironmentsPath()}/assets/${assetIdentifier.getParamValue()}/renditions/${renditionIdentifier.getParamValue()}`;
    }
    viewAssetRendition(assetIdentifier, renditionIdentifier) {
        return `${this.getEnvironmentsPath()}/assets/${assetIdentifier.getParamValue()}/renditions/${renditionIdentifier.getParamValue()}`;
    }
    listSubscriptionProjects() {
        return `${this.getSubscriptionPath()}/projects`;
    }
    listSubscriptionUsers() {
        return `${this.getSubscriptionPath()}/users`;
    }
    viewSubscriptionProject(identifier) {
        return `${this.getSubscriptionPath()}/projects/${identifier.getParamValue()}`;
    }
    viewSubscriptionUser(identifier) {
        return `${this.getSubscriptionPath()}/users/${identifier.getParamValue()}`;
    }
    activateUserInAllProjects(identifier) {
        return `${this.getSubscriptionPath()}/users/${identifier.getParamValue()}/activate`;
    }
    deactivateUserInAllProjects(identifier) {
        return `${this.getSubscriptionPath()}/users/${identifier.getParamValue()}/deactivate`;
    }
    listRoles() {
        return `${this.getEnvironmentsPath()}/roles`;
    }
    viewRole(identifier) {
        return `${this.getEnvironmentsPath()}/roles/${identifier.getParamValue()}`;
    }
    getPreviewConfigruation() {
        return `${this.getEnvironmentsPath()}/preview-configuration`;
    }
    modifyPreviewConfigruation() {
        return `${this.getEnvironmentsPath()}/preview-configuration`;
    }
    inviteProjectUser() {
        return `${this.getEnvironmentsPath()}/users`;
    }
    changeProjectUserRoles(identifier) {
        return `${this.getEnvironmentsPath()}/users/${identifier.getParamValue()}/roles`;
    }
    getEnvironmentCloningStatus() {
        return `${this.getEnvironmentsPath()}/environment-cloning-state`;
    }
    deleteEnvironment() {
        return this.getEnvironmentsPath();
    }
    modifyEnvironment() {
        return this.getEnvironmentsPath();
    }
    cloneEnvironment() {
        return `${this.getEnvironmentsPath()}/clone-environment`;
    }
    markEnvironmentAsProduction() {
        return `${this.getEnvironmentsPath()}/mark-environment-as-production`;
    }
    listSpaces() {
        return `${this.getEnvironmentsPath()}/spaces`;
    }
    viewSpace(identifier) {
        return `${this.getEnvironmentsPath()}/spaces/${identifier.getParamValue()}`;
    }
    addSpace() {
        return `${this.getEnvironmentsPath()}/spaces`;
    }
    deleteSpace(identifier) {
        return `${this.getEnvironmentsPath()}/spaces/${identifier.getParamValue()}`;
    }
    modifySpace(identifier) {
        return `${this.getEnvironmentsPath()}/spaces/${identifier.getParamValue()}`;
    }
    activateWebSpotlight() {
        return `${this.getEnvironmentsPath()}/web-spotlight/activate`;
    }
    deactivateWebSpotlight() {
        return `${this.getEnvironmentsPath()}/web-spotlight/deactivate`;
    }
    checkWebSpotlightStatus() {
        return `${this.getEnvironmentsPath()}/web-spotlight/status`;
    }
    getSubscriptionPath() {
        if (!this.subscriptionId) {
            throw Error(`SubscriptionId was not provided in client configuration`);
        }
        return `${this.subscriptionsPath}/${this.subscriptionId}`;
    }
    getEnvironmentsPath() {
        if (!this.environmentId) {
            throw Error(`EnvironmentId was not provided in client configuration`);
        }
        return `${this.projectsPath}/${this.environmentId}`;
    }
}
