// @kontent-ai/management-sdk/dist/es6/responses/languages/language-responses.js
import { BaseResponses } from '../base-responses';
export var LanguageResponses;
(function (LanguageResponses) {
    class ListLanguagesResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.ListLanguagesResponse = ListLanguagesResponse;
    class ListAllLanguagesResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    LanguageResponses.ListAllLanguagesResponse = ListAllLanguagesResponse;
    class ViewLanguageResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.ViewLanguageResponse = ViewLanguageResponse;
    class AddLanguageResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.AddLanguageResponse = AddLanguageResponse;
    class ModifyLanguageResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.ModifyLanguageResponse = ModifyLanguageResponse;
})(LanguageResponses || (LanguageResponses = {}));
