// @kontent-ai/management-sdk/dist/cjs/responses/asset-renditions/asset-rendition-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetRenditionResponses = void 0;
const base_responses_1 = require("../base-responses");
var AssetRenditionResponses;
(function (AssetRenditionResponses) {
    class AssetRenditionsListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.AssetRenditionsListResponse = AssetRenditionsListResponse;
    class AssetRenditionsListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    AssetRenditionResponses.AssetRenditionsListAllResponse = AssetRenditionsListAllResponse;
    class ModifyAssetRenditionResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.ModifyAssetRenditionResponse = ModifyAssetRenditionResponse;
    class ViewAssetRenditionResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.ViewAssetRenditionResponse = ViewAssetRenditionResponse;
    class AddAssetRenditionResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.AddAssetRenditionResponse = AddAssetRenditionResponse;
})(AssetRenditionResponses || (exports.AssetRenditionResponses = AssetRenditionResponses = {}));
