// @kontent-ai/management-sdk/dist/cjs/responses/content-type-snippets/content-type-snippet-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentTypeSnippetResponses = void 0;
const base_responses_1 = require("../base-responses");
var ContentTypeSnippetResponses;
(function (ContentTypeSnippetResponses) {
    class ContentTypeSnippetListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.ContentTypeSnippetListResponse = ContentTypeSnippetListResponse;
    class ContentTypeSnippetListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    ContentTypeSnippetResponses.ContentTypeSnippetListAllResponse = ContentTypeSnippetListAllResponse;
    class ViewContentTypeSnippetResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.ViewContentTypeSnippetResponse = ViewContentTypeSnippetResponse;
    class AddContentTypeSnippetResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.AddContentTypeSnippetResponse = AddContentTypeSnippetResponse;
    class ModifyContentTypeSnippetResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.ModifyContentTypeSnippetResponse = ModifyContentTypeSnippetResponse;
})(ContentTypeSnippetResponses || (exports.ContentTypeSnippetResponses = ContentTypeSnippetResponses = {}));
