// @kontent-ai/management-sdk/dist/cjs/query-builders/custom-apps-identifier.query.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAppsIdentifierQuery = void 0;
const models_1 = require("../models");
class CustomAppsIdentifierQuery {
    constructor(config, queryService, buildResult) {
        this.config = config;
        this.queryService = queryService;
        this.buildResult = buildResult;
    }
    /**
     * Gets using internal Id
     * @param id Internal Id of custom app
     */
    byCustomAppId(id) {
        return this.buildResult(this.config, this.queryService, new models_1.Identifiers.CustomAppIdentifier(models_1.Identifiers.CustomAppIdentifierEnum.InternalId, id));
    }
    /**
     * Gets query using codename
     * @param codename Codename of custom app
     */
    byCustomAppCodename(codename) {
        return this.buildResult(this.config, this.queryService, new models_1.Identifiers.CustomAppIdentifier(models_1.Identifiers.CustomAppIdentifierEnum.Codename, codename));
    }
}
exports.CustomAppsIdentifierQuery = CustomAppsIdentifierQuery;
