// @kontent-ai/management-sdk/dist/esnext/responses/webhook/webhook-responses.js
import { BaseResponses } from '../base-responses';
export var WebhookResponses;
(function (WebhookResponses) {
    class GetLegacyWebhookResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.GetLegacyWebhookResponse = GetLegacyWebhookResponse;
    class GetWebhookResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.GetWebhookResponse = GetWebhookResponse;
    class AddLegacyWebhookResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.AddLegacyWebhookResponse = AddLegacyWebhookResponse;
    class AddWebhookResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.AddWebhookResponse = AddWebhookResponse;
    class LegacyWebhookListResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.LegacyWebhookListResponse = LegacyWebhookListResponse;
    class WebhookListResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.WebhookListResponse = WebhookListResponse;
})(WebhookResponses || (WebhookResponses = {}));
