// @kontent-ai/management-sdk/dist/cjs/queries/language-variants/filter-language-variants-query.class.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterLanguageVariantsQuery = void 0;
const responses_1 = require("../../responses");
const base_listing_query_1 = require("../base-listing-query");
class FilterLanguageVariantsQuery extends base_listing_query_1.BaseListingQuery {
    constructor(config, queryService, data) {
        super(config, queryService);
        this.config = config;
        this.queryService = queryService;
        this.data = data;
    }
    toPromise() {
        return this.queryService.filterLanguageVariantsAsync(this.getUrl(), this.data, this.queryConfig);
    }
    getAction() {
        return this.apiEndpoints.filterLanguageVariants();
    }
    allResponseFactory(items, responses) {
        return new responses_1.LanguageVariantResponses.ListAllFilterLanguageVariantsResponse({
            items: items,
            responses: responses
        });
    }
}
exports.FilterLanguageVariantsQuery = FilterLanguageVariantsQuery;
