// @kontent-ai/management-sdk/dist/cjs/queries/language-variants/upsert-language-variant-query.class.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertLanguageVariantQuery = void 0;
const language_variant_elements_builder_1 = require("../../models/language-variants/language-variant-elements-builder");
const base_query_1 = require("../base-query");
class UpsertLanguageVariantQuery extends base_query_1.BaseQuery {
    constructor(config, queryService, contentItemIdentifier, languageIdentifier, data) {
        super(config, queryService);
        this.config = config;
        this.queryService = queryService;
        this.contentItemIdentifier = contentItemIdentifier;
        this.languageIdentifier = languageIdentifier;
        this.data = data;
    }
    toPromise() {
        return this.queryService.upsertLanguageVariantAsync(this.getUrl(), this.data(language_variant_elements_builder_1.languageVariantElementsBuilder), this.queryConfig);
    }
    getAction() {
        return this.apiEndpoints.upsertLanguageVariant(this.contentItemIdentifier, this.languageIdentifier);
    }
}
exports.UpsertLanguageVariantQuery = UpsertLanguageVariantQuery;
