// @kontent-ai/management-sdk/dist/cjs/queries/language-variants/view-language-variant-query.class.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewLanguageVariantQuery = void 0;
const base_query_1 = require("../base-query");
class ViewLanguageVariantQuery extends base_query_1.BaseQuery {
    constructor(config, queryService, contentItemIdentifier, languageIdentifier) {
        super(config, queryService);
        this.config = config;
        this.queryService = queryService;
        this.contentItemIdentifier = contentItemIdentifier;
        this.languageIdentifier = languageIdentifier;
        this.fetchPublishedVersion = false;
    }
    toPromise() {
        return this.queryService.viewLanguageVariantAsync(this.getUrl(), this.queryConfig);
    }
    published() {
        this.fetchPublishedVersion = true;
        return this;
    }
    getAction() {
        return this.apiEndpoints.viewLanguageVariant(this.contentItemIdentifier, this.languageIdentifier, this.fetchPublishedVersion);
    }
}
exports.ViewLanguageVariantQuery = ViewLanguageVariantQuery;
