// @kontent-ai/management-sdk/dist/esnext/responses/content-type-snippets/content-type-snippet-responses.js
import { BaseResponses } from '../base-responses';
export var ContentTypeSnippetResponses;
(function (ContentTypeSnippetResponses) {
    class ContentTypeSnippetListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.ContentTypeSnippetListResponse = ContentTypeSnippetListResponse;
    class ContentTypeSnippetListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    ContentTypeSnippetResponses.ContentTypeSnippetListAllResponse = ContentTypeSnippetListAllResponse;
    class ViewContentTypeSnippetResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.ViewContentTypeSnippetResponse = ViewContentTypeSnippetResponse;
    class AddContentTypeSnippetResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.AddContentTypeSnippetResponse = AddContentTypeSnippetResponse;
    class ModifyContentTypeSnippetResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeSnippetResponses.ModifyContentTypeSnippetResponse = ModifyContentTypeSnippetResponse;
})(ContentTypeSnippetResponses || (ContentTypeSnippetResponses = {}));
