// @kontent-ai/management-sdk/dist/es6/responses/content-types/content-type-responses.js
import { BaseResponses } from '../base-responses';
export var ContentTypeResponses;
(function (ContentTypeResponses) {
    class ContentTypeListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.ContentTypeListResponse = ContentTypeListResponse;
    class ContentTypeListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    ContentTypeResponses.ContentTypeListAllResponse = ContentTypeListAllResponse;
    class ModifyContentTypeResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.ModifyContentTypeResponse = ModifyContentTypeResponse;
    class ViewContentTypeResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.ViewContentTypeResponse = ViewContentTypeResponse;
    class AddContentTypeResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.AddContentTypeResponse = AddContentTypeResponse;
})(ContentTypeResponses || (ContentTypeResponses = {}));
