// @kontent-ai/management-sdk/dist/cjs/responses/webhook/webhook-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookResponses = void 0;
const base_responses_1 = require("../base-responses");
var WebhookResponses;
(function (WebhookResponses) {
    class GetLegacyWebhookResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.GetLegacyWebhookResponse = GetLegacyWebhookResponse;
    class GetWebhookResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.GetWebhookResponse = GetWebhookResponse;
    class AddLegacyWebhookResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.AddLegacyWebhookResponse = AddLegacyWebhookResponse;
    class AddWebhookResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.AddWebhookResponse = AddWebhookResponse;
    class LegacyWebhookListResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.LegacyWebhookListResponse = LegacyWebhookListResponse;
    class WebhookListResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    WebhookResponses.WebhookListResponse = WebhookListResponse;
})(WebhookResponses || (exports.WebhookResponses = WebhookResponses = {}));
