// @kontent-ai/management-sdk/dist/esnext/responses/content-items/content-item-responses.js
import { BaseResponses } from '../base-responses';
export var ContentItemResponses;
(function (ContentItemResponses) {
    class ContentItemsResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.ContentItemsResponse = ContentItemsResponse;
    class ContentItemsListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    ContentItemResponses.ContentItemsListAllResponse = ContentItemsListAllResponse;
    class ViewContentItemResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.ViewContentItemResponse = ViewContentItemResponse;
    class AddContentItemResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.AddContentItemResponse = AddContentItemResponse;
    class UpdateContentItemResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.UpdateContentItemResponse = UpdateContentItemResponse;
    class UpsertContentItemResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentItemResponses.UpsertContentItemResponse = UpsertContentItemResponse;
})(ContentItemResponses || (ContentItemResponses = {}));
