// @kontent-ai/management-sdk/dist/esnext/responses/subscriptions/subscription-responses.js
import { BaseResponses } from '../base-responses';
export var SubscriptionResponses;
(function (SubscriptionResponses) {
    class SubscriptionProjectsListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.SubscriptionProjectsListResponse = SubscriptionProjectsListResponse;
    class SubscriptionProjectsListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    SubscriptionResponses.SubscriptionProjectsListAllResponse = SubscriptionProjectsListAllResponse;
    class ViewSubscriptionProjectResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.ViewSubscriptionProjectResponse = ViewSubscriptionProjectResponse;
    class SubscriptionUsersListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.SubscriptionUsersListResponse = SubscriptionUsersListResponse;
    class SubscriptionUsersListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    SubscriptionResponses.SubscriptionUsersListAllResponse = SubscriptionUsersListAllResponse;
    class ViewSubscriptionUserResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.ViewSubscriptionUserResponse = ViewSubscriptionUserResponse;
})(SubscriptionResponses || (SubscriptionResponses = {}));
