// @kontent-ai/management-sdk/dist/cjs/responses/subscriptions/subscription-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionResponses = void 0;
const base_responses_1 = require("../base-responses");
var SubscriptionResponses;
(function (SubscriptionResponses) {
    class SubscriptionProjectsListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.SubscriptionProjectsListResponse = SubscriptionProjectsListResponse;
    class SubscriptionProjectsListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    SubscriptionResponses.SubscriptionProjectsListAllResponse = SubscriptionProjectsListAllResponse;
    class ViewSubscriptionProjectResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.ViewSubscriptionProjectResponse = ViewSubscriptionProjectResponse;
    class SubscriptionUsersListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.SubscriptionUsersListResponse = SubscriptionUsersListResponse;
    class SubscriptionUsersListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    SubscriptionResponses.SubscriptionUsersListAllResponse = SubscriptionUsersListAllResponse;
    class ViewSubscriptionUserResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    SubscriptionResponses.ViewSubscriptionUserResponse = ViewSubscriptionUserResponse;
})(SubscriptionResponses || (exports.SubscriptionResponses = SubscriptionResponses = {}));
