// @kontent-ai/management-sdk/dist/es6/responses/custom-apps/custom-apps.responses.js
import { BaseResponses } from '../base-responses';
export var CustomAppsResponses;
(function (CustomAppsResponses) {
    class AddCustomAppResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    CustomAppsResponses.AddCustomAppResponse = AddCustomAppResponse;
    class CustomAppsListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    CustomAppsResponses.CustomAppsListResponse = CustomAppsListResponse;
    class CustomAppsListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    CustomAppsResponses.CustomAppsListAllResponse = CustomAppsListAllResponse;
    class ModifyCustomAppResponse extends AddCustomAppResponse {
    }
    CustomAppsResponses.ModifyCustomAppResponse = ModifyCustomAppResponse;
    class GetCustomAppResponse extends AddCustomAppResponse {
    }
    CustomAppsResponses.GetCustomAppResponse = GetCustomAppResponse;
})(CustomAppsResponses || (CustomAppsResponses = {}));
