// @kontent-ai/management-sdk/dist/cjs/responses/custom-apps/custom-apps.responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAppsResponses = void 0;
const base_responses_1 = require("../base-responses");
var CustomAppsResponses;
(function (CustomAppsResponses) {
    class AddCustomAppResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    CustomAppsResponses.AddCustomAppResponse = AddCustomAppResponse;
    class CustomAppsListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    CustomAppsResponses.CustomAppsListResponse = CustomAppsListResponse;
    class CustomAppsListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    CustomAppsResponses.CustomAppsListAllResponse = CustomAppsListAllResponse;
    class ModifyCustomAppResponse extends AddCustomAppResponse {
    }
    CustomAppsResponses.ModifyCustomAppResponse = ModifyCustomAppResponse;
    class GetCustomAppResponse extends AddCustomAppResponse {
    }
    CustomAppsResponses.GetCustomAppResponse = GetCustomAppResponse;
})(CustomAppsResponses || (exports.CustomAppsResponses = CustomAppsResponses = {}));
