// @kontent-ai/management-sdk/dist/cjs/models/language-variants/language-variant-elements-builder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.languageVariantElementsBuilder = exports.LanguageVariantElementsBuilder = void 0;
class LanguageVariantElementsBuilder {
    assetElement(element) {
        return element;
    }
    customElement(element) {
        return element;
    }
    dateTimeElement(element) {
        return element;
    }
    linkedItemsElement(element) {
        return element;
    }
    multipleChoiceElement(element) {
        return element;
    }
    numberElement(element) {
        return element;
    }
    richTextElement(element) {
        return element;
    }
    taxonomyElement(element) {
        return element;
    }
    textElement(element) {
        return element;
    }
    urlSlugElement(element) {
        return element;
    }
    any(element) {
        return element;
    }
}
exports.LanguageVariantElementsBuilder = LanguageVariantElementsBuilder;
exports.languageVariantElementsBuilder = new LanguageVariantElementsBuilder();
