// @kontent-ai/management-sdk/dist/es6/responses/asset-renditions/asset-rendition-responses.js
import { BaseResponses } from '../base-responses';
export var AssetRenditionResponses;
(function (AssetRenditionResponses) {
    class AssetRenditionsListResponse extends BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.AssetRenditionsListResponse = AssetRenditionsListResponse;
    class AssetRenditionsListAllResponse extends BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    AssetRenditionResponses.AssetRenditionsListAllResponse = AssetRenditionsListAllResponse;
    class ModifyAssetRenditionResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.ModifyAssetRenditionResponse = ModifyAssetRenditionResponse;
    class ViewAssetRenditionResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.ViewAssetRenditionResponse = ViewAssetRenditionResponse;
    class AddAssetRenditionResponse extends BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    AssetRenditionResponses.AddAssetRenditionResponse = AddAssetRenditionResponse;
})(AssetRenditionResponses || (AssetRenditionResponses = {}));
