// @kontent-ai/management-sdk/dist/cjs/responses/environments/environment-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentResponses = void 0;
const base_responses_1 = require("../base-responses");
var EnvironmentResponses;
(function (EnvironmentResponses) {
    class EnvironmentValidationIssuesListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.EnvironmentValidationIssuesListResponse = EnvironmentValidationIssuesListResponse;
    class EnvironmentValidationIssuesListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    EnvironmentResponses.EnvironmentValidationIssuesListAllResponse = EnvironmentValidationIssuesListAllResponse;
    class StartEnvironmentValidationResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.StartEnvironmentValidationResponse = StartEnvironmentValidationResponse;
    class CheckEnvironmentValidationResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.CheckEnvironmentValidationResponse = CheckEnvironmentValidationResponse;
    class EnvironmentInformationResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.EnvironmentInformationResponse = EnvironmentInformationResponse;
    class GetCloningStateResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.GetCloningStateResponse = GetCloningStateResponse;
    class ModifyEnvironmentResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.ModifyEnvironmentResponse = ModifyEnvironmentResponse;
    class CloneEnvironmentResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    EnvironmentResponses.CloneEnvironmentResponse = CloneEnvironmentResponse;
})(EnvironmentResponses || (exports.EnvironmentResponses = EnvironmentResponses = {}));
