// @kontent-ai/management-sdk/dist/cjs/responses/content-types/content-type-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentTypeResponses = void 0;
const base_responses_1 = require("../base-responses");
var ContentTypeResponses;
(function (ContentTypeResponses) {
    class ContentTypeListResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.ContentTypeListResponse = ContentTypeListResponse;
    class ContentTypeListAllResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    ContentTypeResponses.ContentTypeListAllResponse = ContentTypeListAllResponse;
    class ModifyContentTypeResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.ModifyContentTypeResponse = ModifyContentTypeResponse;
    class ViewContentTypeResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.ViewContentTypeResponse = ViewContentTypeResponse;
    class AddContentTypeResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    ContentTypeResponses.AddContentTypeResponse = AddContentTypeResponse;
})(ContentTypeResponses || (exports.ContentTypeResponses = ContentTypeResponses = {}));
