// @kontent-ai/management-sdk/dist/cjs/mappers/custom-app-mapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.customAppMapper = exports.CustomAppMapper = void 0;
const models_1 = require("../models");
const responses_1 = require("../responses");
const base_mapper_1 = require("./base-mapper");
class CustomAppMapper extends base_mapper_1.BaseMapper {
    mapGetCustomAppResponse(response) {
        return new responses_1.CustomAppsResponses.GetCustomAppResponse(super.mapResponseDebug(response), response.data, this.mapCustomApp(response.data));
    }
    mapListCustomAppsResponse(response) {
        return new responses_1.CustomAppsResponses.CustomAppsListResponse(super.mapResponseDebug(response), response.data, {
            pagination: super.mapPagination(response.data.pagination),
            items: response.data.custom_apps.map((m) => this.mapCustomApp(m))
        });
    }
    mapModifyCustomAppResponse(response) {
        return new responses_1.CustomAppsResponses.ModifyCustomAppResponse(super.mapResponseDebug(response), response.data, this.mapCustomApp(response.data));
    }
    mapAddCustomAppResponse(response) {
        return new responses_1.CustomAppsResponses.AddCustomAppResponse(super.mapResponseDebug(response), response.data, this.mapCustomApp(response.data));
    }
    mapCustomApp(rawCustomApp) {
        return new models_1.CustomAppModels.CustomApp(Object.assign(Object.assign({}, rawCustomApp), { _raw: rawCustomApp }));
    }
}
exports.CustomAppMapper = CustomAppMapper;
exports.customAppMapper = new CustomAppMapper();
