// @kontent-ai/management-sdk/dist/cjs/responses/languages/language-responses.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LanguageResponses = void 0;
const base_responses_1 = require("../base-responses");
var LanguageResponses;
(function (LanguageResponses) {
    class ListLanguagesResponse extends base_responses_1.BaseResponses.BaseContentManagementListResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.ListLanguagesResponse = ListLanguagesResponse;
    class ListAllLanguagesResponse extends base_responses_1.BaseResponses.ContentManagementListAllResponse {
        constructor(data) {
            super(data);
        }
    }
    LanguageResponses.ListAllLanguagesResponse = ListAllLanguagesResponse;
    class ViewLanguageResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.ViewLanguageResponse = ViewLanguageResponse;
    class AddLanguageResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.AddLanguageResponse = AddLanguageResponse;
    class ModifyLanguageResponse extends base_responses_1.BaseResponses.BaseContentManagementResponse {
        constructor(debug, rawData, data) {
            super(debug, rawData, data);
        }
    }
    LanguageResponses.ModifyLanguageResponse = ModifyLanguageResponse;
})(LanguageResponses || (exports.LanguageResponses = LanguageResponses = {}));
